### DISCUSSION TITLE
Dev toolchains of Jamulus team ?

### DISCUSSION THREAD

**Discussion by pcar75** [jamulussoftware/jamulus#2977] [General]
Bonjour,
I've restarted programming in C and learning C++.  If / when I feel proficient enough, I may offer to help the Jamulus team. However, I find dev environments are a lot more complex than decades ago.
For local dev , what are the tools (compiler, IDE, libraries/framework , etc. ) your team mostly use ?

Context : dual-boot pc
Windows 10 ; mingw64 compilers 10.0.0 (GCC12.1.0) , Eclipse Java and C/C++ 22-09 , Clang-format 14.0.0 (run outside of Eclipse :-/) , 
Ubuntu Studio 22.04 LTS : dev env to be configure soon.

Thanks for suggestions.

**Comment by ann0see** [comment]
Hi @pcar75,

Thanks for trying to help out. I think most people of the dev team work on Linux. The setup experience is way better than Windows. 
I'm using Atom (+ trying to get used to vim) and compile via command line. 

@pgScorpio uses Windows but it's painful to setup.

MacOS has a bug with the direct Xcode method, so compiling via deploy_mac.ah is needed until we fix this bug but you don't seem to be affected.

**Comment by ann0see** [comment]
I've heard good things about VSCode, so maybe it's worth looking into it. 
QT also provides an IDE, but that's not that easy.

**Comment by pljones** [comment]
For Windows and for the UI, I use Qt (as my main desktop is Windows). I run Qt over remote X sometimes if I want to see if things look different on Qt on Linux.  Generally (non-UI), I'll use vi + running `make distclean; qmake && make && make clean` -- plus attention to detail.  When making small changes, I'll probably add lots of debugging output and just use `make`, checking each change has had the expected effect.

**Reply by hoffie** [reply]
> Generally (non-UI), I'll use vi + running `make distclean; qmake && make && make clean` -- plus attention to detail. When making small changes, I'll probably add lots of debugging output and just use `make`, checking each change has had the expected effect.

Same here.

**Comment by dcorson-ticino-com** [comment]
I run Windows and use the QT environment, easy peasy.
Not sure why the Linux friends claim it is difficult.

**Reply by ann0see** [reply]
Probably it depends on what you're comfortable with 😀

**Comment by pcar75** [comment]
Thanks all for comments so far.
@ann0see I'm comfortable on a command-line (WIN-cmd or Linux) ,  I thought of trying Ec;lipse because it's open , can support many languages and is on many platforms so I won't have to relearn IDE when switching between these.  However I find myself spending almost more time figuring out Eclipse than programming :-/

Many friends use VS or VC in Windows C# , but I'm not sure I trust Microsoft about the breadth and quality of their support on C/C++ and Java, and  non-Windows platforms in general.

**Reply by ann0see** [reply]
> However I find myself spending almost more time figuring out Eclipse than programming :-/

I use Eclipse for Java related programming too, but concluded that it can get very confusing sometimes. I wouldn't use it if you don't feel comfortable with it and try something else like VSCode. At least to some small extent (I switched over to VIM after some time and didn't have much time to try it fully), I think it's fine to use it on Linux for C programming. I didn't try it for C++ but as it's very extensible I'd suggest to install it, play around with some features maybe read some articles or watch videos about the setup etc.
